plugins/discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/AwsEc2Utils.java

```diff
@@ -24,7 +26,8 @@
 class AwsEc2Utils {
 
     private static final Logger logger = LogManager.getLogger(AwsEc2Utils.class);
-    private static final int CONNECT_TIMEOUT = 2000;
+    // The timeout can be configured via the AWS_METADATA_SERVICE_TIMEOUT environment variable
+    private static final int TIMEOUT = ConnectionUtils.getInstance().getTimeoutMillis();
```

Comment:
I see the DEFAULT_TIMEOUT_MILLIS in the SDK is 1000, which is different than the previous 2000 timeout we had. Is that concerning? Should we better detect if the env var is not set, and set our own 2000ms default value, to keep the old behavior?